### ISSUE TITLE
SB1: Suspend not working anymore in 5.14

### ISSUE THREAD

**Issue by OliverUrbann** [linux-surface/linux-surface#585]
Pressing the power button seems to activate the suspend mode but wake up by pressing again does not seem to work, at least the display is not activated again. It worked with 5.13.13.

### Environment

- Hardware model: Surface Book 1
- Kernel version: Linux ciVicSurface 5.14.8-surface #1 SMP Mon Sep 27 15:43:15 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux
- Distribution: Ubuntu 20.04 LTS

**Comment by qzed** [comment]
Hmm, works fine on my SB2 so might be device specific or some user-space issue. Not sure how well that works, but on the SB2 you can use the keyboard backlight/Fn indicator light to see where it's stuck:
- if it stays on, this would suggest it's not even suspending properly
- it turns off and turns back on after suspend, it's either an issue in late resume or user-space
- if turns off but doesn't turn back on it's an issue in either late suspend or early resume, somewhere in the kernel

Also maybe swipe on the touchpad once or twice to make sure it's not Gnome/your DE turning off the display early, that has lead to some confusion before, I think.

**Comment by OliverUrbann** [comment]
Thx so far. Even with 5.13 the backlight stays on, so actually is also not working with the old kernel? Here is a dmesg from this "working" suspend:

```
[  403.823078] mwifiex_pcie 0000:03:00.0: info: successfully disconnected from 18:83:bf:66:bd:25: reason code 3
[  404.267618] PM: suspend entry (s2idle)
[  404.307675] Filesystems sync: 0.040 seconds
[  404.308136] Freezing user space processes ... (elapsed 0.002 seconds) done.
[  404.310446] OOM killer disabled.
[  404.310447] Freezing remaining freezable tasks ... (elapsed 0.001 seconds) done.
[  404.311722] printk: Suspending console(s) (use no_console_suspend to debug)
[  404.369521] intel_pch_thermal 0000:00:14.2: CPU-PCH is cool [30C], continue to suspend
[  404.380182] ipts 0000:00:16.4-3e8d0870-271a-4208-8eb5-9acb9402ae04: Stopping IPTS
[  404.530328] ACPI: EC: interrupt blocked
[  415.355877] ACPI: EC: interrupt unblocked
[  415.621076] nvme nvme0: 4/0/0 default/read/poll queues
[  415.858306] OOM killer enabled.
[  415.858310] Restarting tasks ... 
[  415.858386] ipts 0000:00:16.4-3e8d0870-271a-4208-8eb5-9acb9402ae04: Starting IPTS
[  415.880696] done.
[  415.882523] ipts 0000:00:16.4-3e8d0870-271a-4208-8eb5-9acb9402ae04: Device 1B96:005E ready
[  416.058092] PM: suspend exit
```

**Comment by qzed** [comment]
> Even with 5.13 the backlight stays on, so actually is also not working with the old kernel?

Might be a SB1 thing, I'm not really sure if that can be compared to the SB2 because the SB1 doesn't have support for anything EC related yet. The logs don't show any errors so I assume it's working correctly on v5.13 and the backlight behaves different on SB1 vs. SB2. Bit unfortunate, because that makes tracking down the issue harder.

**Comment by OliverUrbann** [comment]
I tried the standby connected to the Surface Dock. Surprisingly it works and I can see that the keyboard light is switched off immediately with 5.14. With 5.13 the light stays on. So I assume something related to standby has changed from 5.13.13 to 5.14.8.

**Comment by qzed** [comment]
Huh, dock or no dock shouldn't make any difference, that's basically just a bunch of USB ports. Do the lights stay off when you swipe on the touchpad or press any keys?

**Comment by OliverUrbann** [comment]
I tracked it down to the power supply. With just the PSU, the keyboard light stays on for a few seconds (usual timeout I think). Fn works (LED on/off, seems to be handled by firmware), caps lock does not work (LED stays off), using the touchpad does not turn on keyboard light, and WiFi is off (I can see it reconnecting after wake-up). Here is a dmesg:

```
[   68.109616] mwifiex_pcie 0000:03:00.0: info: successfully disconnected from 18:83:bf:66:bd:25: reason code 3
[   68.363003] PM: suspend entry (s2idle)
[   68.390168] Filesystems sync: 0.027 seconds
[   68.390362] Freezing user space processes ... (elapsed 0.785 seconds) done.
[   69.175783] OOM killer disabled.
[   69.175787] Freezing remaining freezable tasks ... (elapsed 0.001 seconds) done.
[   69.177693] printk: Suspending console(s) (use no_console_suspend to debug)
[   69.236175] intel_pch_thermal 0000:00:14.2: CPU-PCH is cool [32C], continue to suspend
[   69.247420] ipts 0000:00:16.4-3e8d0870-271a-4208-8eb5-9acb9402ae04: Stopping IPTS
[   69.535209] ACPI: EC: interrupt blocked
[   74.917875] ACPI: EC: interrupt unblocked
[   74.918571] acpi device:45: Cannot transition to power state D0 for parent in D3hot
[   74.918583] acpi device:4c: Cannot transition to power state D0 for parent in D3hot
[   74.961009] acpi device:45: Cannot transition to power state D0 for parent in D3hot
[   74.970356] nvme nvme0: 4/0/0 default/read/poll queues
[   75.260688] OOM killer enabled.
[   75.260694] Restarting tasks ... 
[   75.260725] ipts 0000:00:16.4-3e8d0870-271a-4208-8eb5-9acb9402ae04: Starting IPTS
[   75.264289] done.
[   75.270339] ipts 0000:00:16.4-3e8d0870-271a-4208-8eb5-9acb9402ae04: Device 1B96:005E ready
[   75.479533] PM: suspend exit
[   78.825219] mwifiex_pcie 0000:03:00.0: info: trying to associate to bssid 18:83:bf:66:bd:25
[   78.846629] mwifiex_pcie 0000:03:00.0: info: associated to bssid 18:83:bf:66:bd:25 successfully
[   78.862285] IPv6: ADDRCONF(NETDEV_CHANGE): wlp3s0: link becomes ready
```

**Comment by qzed** [comment]
Okay, so the logs look good. Nothing out of the ordinary there. The `Cannot transition to power state D0 for parent in D3hot` stick a bit out, but we've seen them on a bunch other devices with working suspend and they should (if at all) only affect some devices after resuming. So nothing that should crash/prevent resuming.

I still can't quite make sense of the PSU being the deciding factor. Normally, that should be independent of any other drivers. Is there any chance that systemd/userspace does something different when the PSU is connected vs. when it is not?

What's also interesting is that the Fn led still toggles but the keyboard backlight stays off. On the SB2 those are connected, so if the EC is switched into "display-off"-mode both are turned off. Seems like on the SB1 the backlight might not be controlled by the EC. Also yeah, pretty sure that the Fn thing is handled by the EC (on the SB2 and newer devices at least it doesn't generate any input events). Caps lock is actually an input event (with the led handled by the kernel HID subsystem) so it makes sense that, if the device is (at least partly) suspended, its led doesn't toggle.

Again, I can't really think of how the PSU being connected or disconnected could directly affect things in-kernel (except for the ACPI AC/battery drivers which should be quite solid), but you could try unloading modules. In the past `ipts` and `mwifiex_pcie`/`mwifiex` have made some problems, so you could start with those.

Also running `journalctl -b -1` on the next boot right after the suspend/resume failure might give you some clue as to what's going wrong. (the `-b -1` means "select log from boot `-1`", i.e. last boot). However, if it's an in-kernel issue that likely doesn't show anything.

There are a couple of methods to debug suspend/resume issues, like only freezing processes/going through a reduced number of stages etc. Some info on that can be found at:
- https://www.kernel.org/doc/html/latest/power/basic-pm-debugging.html
- https://01.org/blogs/rzhang/2015/best-practice-debug-linux-suspend/hibernate-issues
- https://wiki.ubuntu.com/DebuggingKernelSuspend

Keep in mind though that the Surface devices use s2idle and not s2ram/S3.

**Comment by OliverUrbann** [comment]
I'm sorry, maybe I wasn't clear enough. The keyboard lights turns on if any key is pressed but not if the touchpad is used.

Thx for the links, I will see what I can find out.

**Comment by qzed** [comment]
Ahh sorry, yeah I think the keyboard is a red herring then.

**Comment by OliverUrbann** [comment]
Thx again for the links. I tried the test procedure writing `freeze`, `devices` and `platform` to `/sys/power/pm_test` where only `platform` fails. According to [kernel.org](https://www.kernel.org/doc/Documentation/power/basic-pm-debugging.txt) this could related to ACPI. 

Furthermore, I also got the log from a failed s2idle with `journalctl`:
```
Okt 03 23:07:40 ciVicSurface NetworkManager[1311]: <info>  [1633295260.1252] manager: sleep: sleep requested (sleeping: no  enabled: yes)
Okt 03 23:07:40 ciVicSurface NetworkManager[1311]: <info>  [1633295260.1254] device (p2p-dev-wlp3s0): state change: disconnected -> unmanaged (reason 'sleeping', sys-iface-state: >
Okt 03 23:07:40 ciVicSurface NetworkManager[1311]: <info>  [1633295260.1260] manager: NetworkManager state is now ASLEEP
Okt 03 23:07:40 ciVicSurface NetworkManager[1311]: <info>  [1633295260.1263] device (wlp3s0): state change: activated -> deactivating (reason 'sleeping', sys-iface-state: 'managed>
Okt 03 23:07:40 ciVicSurface ModemManager[1414]: <info>  [sleep-monitor] system is about to suspend
Okt 03 23:07:40 ciVicSurface whoopsie[1926]: [23:07:40] offline
Okt 03 23:07:40 ciVicSurface dbus-daemon[1309]: [system] Activating via systemd: service name='org.freedesktop.nm_dispatcher' unit='dbus-org.freedesktop.nm-dispatcher.service' req>
Okt 03 23:07:40 ciVicSurface systemd[1]: Starting Network Manager Script Dispatcher Service...
Okt 03 23:07:40 ciVicSurface dbus-daemon[1309]: [system] Successfully activated service 'org.freedesktop.nm_dispatcher'
Okt 03 23:07:40 ciVicSurface systemd[1]: Started Network Manager Script Dispatcher Service.
Okt 03 23:07:40 ciVicSurface wpa_supplicant[1351]: wlp3s0: CTRL-EVENT-DISCONNECTED bssid=18:83:bf:66:bd:25 reason=3 locally_generated=1
Okt 03 23:07:40 ciVicSurface kernel: mwifiex_pcie 0000:03:00.0: info: successfully disconnected from 18:83:bf:66:bd:25: reason code 3
Okt 03 23:07:40 ciVicSurface NetworkManager[1311]: <warn>  [1633295260.1753] sup-iface[0x558c9e34a120,wlp3s0]: connection disconnected (reason -3)
Okt 03 23:07:40 ciVicSurface NetworkManager[1311]: <info>  [1633295260.1795] device (wlp3s0): state change: deactivating -> disconnected (reason 'sleeping', sys-iface-state: 'mana>
Okt 03 23:07:40 ciVicSurface avahi-daemon[1304]: Withdrawing address record for fe80::9810:4bcb:8287:4ba3 on wlp3s0.
Okt 03 23:07:40 ciVicSurface avahi-daemon[1304]: Leaving mDNS multicast group on interface wlp3s0.IPv6 with address fe80::9810:4bcb:8287:4ba3.
Okt 03 23:07:40 ciVicSurface avahi-daemon[1304]: Interface wlp3s0.IPv6 no longer relevant for mDNS.
Okt 03 23:07:40 ciVicSurface whoopsie[1926]: [23:07:40] Cannot reach: https://daisy.ubuntu.com
Okt 03 23:07:40 ciVicSurface NetworkManager[1311]: <info>  [1633295260.2091] dhcp4 (wlp3s0): canceled DHCP transaction
Okt 03 23:07:40 ciVicSurface NetworkManager[1311]: <info>  [1633295260.2091] dhcp4 (wlp3s0): state changed bound -> done
Okt 03 23:07:40 ciVicSurface avahi-daemon[1304]: Withdrawing address record for 192.168.0.132 on wlp3s0.
Okt 03 23:07:40 ciVicSurface gnome-shell[2448]: An active wireless connection, in infrastructure mode, involves no access point?
Okt 03 23:07:40 ciVicSurface avahi-daemon[1304]: Leaving mDNS multicast group on interface wlp3s0.IPv4 with address 192.168.0.132.
Okt 03 23:07:40 ciVicSurface avahi-daemon[1304]: Interface wlp3s0.IPv4 no longer relevant for mDNS.
Okt 03 23:07:40 ciVicSurface NetworkManager[1311]: <info>  [1633295260.2184] device (wlp3s0): supplicant interface state: completed -> disconnected
Okt 03 23:07:40 ciVicSurface NetworkManager[1311]: <info>  [1633295260.2190] device (wlp3s0): state change: disconnected -> unmanaged (reason 'sleeping', sys-iface-state: 'managed>
Okt 03 23:07:40 ciVicSurface wpa_supplicant[1351]: wlp3s0: CTRL-EVENT-SCAN-FAILED ret=-100
Okt 03 23:07:40 ciVicSurface systemd[1]: Reached target Sleep.
Okt 03 23:07:40 ciVicSurface systemd[1]: Starting Record successful boot for GRUB...
Okt 03 23:07:40 ciVicSurface systemd[1]: Starting Suspend...
Okt 03 23:07:40 ciVicSurface systemd[1]: grub-common.service: Succeeded.
Okt 03 23:07:40 ciVicSurface systemd[1]: Finished Record successful boot for GRUB.
Okt 03 23:07:40 ciVicSurface systemd[1]: Starting GRUB failed boot detection...
Okt 03 23:07:40 ciVicSurface wpa_supplicant[1351]: nl80211: deinit ifname=wlp3s0 disabled_11b_rates=0
Okt 03 23:07:40 ciVicSurface systemd[1]: grub-initrd-fallback.service: Succeeded.
Okt 03 23:07:40 ciVicSurface systemd[1]: Finished GRUB failed boot detection.
Okt 03 23:07:40 ciVicSurface systemd-sleep[3948]: Suspending system...
Okt 03 23:07:40 ciVicSurface kernel: PM: suspend entry (s2idle)
```

From what I can see, it does not show an error. So ACPI is a somewhat likely cause. What could I do now? It will be probably very hard for me to further investigate it, maybe posting a bug report at some other kernel repos related to that?

**Comment by qzed** [comment]
> Thx again for the links. I tried the test procedure writing `freeze`, `devices` and `platform` to `/sys/power/pm_test` where only `platform` fails. According to [kernel.org](https://www.kernel.org/doc/Documentation/power/basic-pm-debugging.txt) this could related to ACPI.

IIRC if `devices` works, then the device driver code should run through all the necessary steps without crashing. So I think it's very unlikely that it's caused by any crash in the kernel code then... which probably makes this even harder to figure out.

The log unfortunately, as suspected, doesn't show much. It shows suspend entry (last line), but nothing after. So seems to enter suspend fine which fits to the `device` test working.

Did you try running `rtcwake` yet? Could be that everything's running fine and it's just a matter of not receiving the wakeup signal. You could also check if the [mainline kernels](https://kernel.ubuntu.com/~kernel-ppa/mainline/) have the same issue and in what release that happens first.

> I think there are no changes related to that in this repo since 5.13.13, right?

No changes that I could think of. All of our changes lately only target AMD hardware.

**Comment by OliverUrbann** [comment]
Ok, rtcwake also does not work. Additionally, I've tested mainline 5.13.19, 5.14-rc2 (rc1 is not available) and 5.15-rc3. The latest working is 5.13.19. What would be the best way to check what changed between 5.13.19 and 5.14-rc2?

**Comment by qzed** [comment]
Okay, so you'll have to look for a change between v5.13.9 and v5.14-rc2 in the upstream kernel.

You can get all changes to that via `git diff v5.13.19..v5.14-rc2 [path]` on the kernel repo, where `[path]` is an optional path inside the repo. That's probably too large to look through without any strategy though, since it's over major versions. Meaning over the merge window (which ends with a new `-rc1`).

What's usually done for such issues is bisecting. That requires you to build the kernel a couple of times though, so will take some time. This process is essentially a binary search for the commit that breaks it. You can start that with `git bisect start` and mark commits as good or bad with `git bisect good` or `git bisect bad`. On each new step you'll have to build the kernel to figure out if it's a good or bad commit. [Here](https://git-scm.com/docs/git-bisect) is some documentation on that. Luckily the SB1 doesn't really need any patches to work (if you can accept bad wifi performance for testing), so you don't have to deal with that as well. Note that you'll need to specify `v5.13` as first good commit since that's the first common ancestor of `v5.13.19` and `v5.14-rc2` (that's where the stable tree starts to diverge). If you want to go down that road and have any questions, feel free to ask.

**Comment by OliverUrbann** [comment]
Ok, I have an intermediate result. It seems that

- Ubuntu mainline kernel 5.13.13 has a .config that works
- Ubuntu mainline kernel 5.14-rc2 has a .config that *does not* work

Doing the following on the kernel.org repo (git://git.kernel.org/pub/scm/linux/kernel/git/stable/linux-stable.git)

```
git checkout v5.13.13
cp config-5.14.8-surface .config && make olddefconfig && make -j48 bindeb-pkg LOCALVERSION=-rc2test5
```

leads do a non working kernel. This is also true for `git checkout v5.14-rc2`.

So my conclusion is, the non working code is already contained in 5.13 but the configuration was change from 5.13 to 5.14 in the Ubuntu mainline kernels. Unfortunately I think a binary search on the config switches is not possible but there are also not much differences, see the attached config files.

[config-5.13.zip](https://github.com/linux-surface/linux-surface/files/7306944/config-5.13.zip)

**Comment by qzed** [comment]
Okay, so below is the diff between `config-5.13.13-surface` and  `config-5.13.13-rc2test5`. Does really not look like much: GCC/LD version stuff. Some `UBSAN` (undefined behavior sanitizer) which we can likely ignore. Same with `KASAN` (kernel address sanitizer) and other test options. Binder/ASHMEM also comes along with four options but that's being disabled in the non-working kernel so also something we can ignore. I spot some changes related to camera patches (`MFD_TPS68470`, `VIDEO_OV5693`, `INTEL_SKL_INT3472`), but since it's also the Surface kernel that works I don't think those should be problematic. Similarly IPTS and the SB1 dGPU switch.

I think that should leave (diff working to non-working):
```diff
-AS_TPAUSE y
-COMMON_CLK_S2MPS11 m
-DRM_I915_GVT_KVMGT m
-GPIO_TPS68470 y
-KPC2000_CORE m
-KPC2000_DMA m
-KPC2000_I2C m
-KPC2000_SPI m
-MAX_RAW_DEVS 256
-REGULATOR_S2MPA01 m
-REGULATOR_S2MPS11 m
-REGULATOR_S5M8767 m
-RTC_DRV_S5M m
 CAIF_HSI m -> n
 HARDENED_USERCOPY_FALLBACK y -> n
 HP_WIRELESS m -> n
 INFINIBAND_I40IW m -> n
 KPC2000 y -> n
 MANDATORY_FILE_LOCKING y -> n
 MFD_SEC_CORE y -> n
 PRISM54 n -> m
 RAW_DRIVER m -> n
 SND_SOC_INTEL_SKYLAKE_HDAUDIO_CODEC n -> y
 SND_SOC_ZX_AUD96P22 m -> n
 VFIO_MDEV_DEVICE m -> n
```

needing closer inspection.

Full diff:

<details>
<summary><code>./scripts/diffconfig config-5.13.13-surface config-5.13.13-rc2test5</code></summary>

```diff
-ANDROID_BINDERFS n
-ANDROID_BINDER_DEVICES "binder"
-ANDROID_BINDER_IPC_SELFTEST n
-AS_TPAUSE y
-CC_HAS_WORKING_NOSANITIZE_ADDRESS y
-COMMON_CLK_S2MPS11 m
-DRM_I915_GVT_KVMGT m
-GPIO_TPS68470 y
-INTEL_SKL_INT3472 m
-KASAN n
-KPC2000_CORE m
-KPC2000_DMA m
-KPC2000_I2C m
-KPC2000_SPI m
-MAX_RAW_DEVS 256
-MISC_IPTS m
-REGULATOR_S2MPA01 m
-REGULATOR_S2MPS11 m
-REGULATOR_S5M8767 m
-RTC_DRV_S5M m
-SURFACE_BOOK1_DGPU_SWITCH m
-TPS68470_PMIC_OPREGION y
-VIDEO_OV5693 m
 ANDROID_BINDER_IPC m -> n
 ASHMEM m -> n
 AS_VERSION 23400 -> 23000
 CAIF_HSI m -> n
 CC_VERSION_TEXT "gcc (Ubuntu 9.3.0-17ubuntu1~20.04) 9.3.0" -> "gcc (Ubuntu 7.5.0-3ubuntu1~18.04) 7.5.0"
 GCC_VERSION 90300 -> 70500
 HARDENED_USERCOPY_FALLBACK y -> n
 HP_WIRELESS m -> n
 INFINIBAND_I40IW m -> n
 KPC2000 y -> n
 LD_VERSION 23400 -> 23000
 MANDATORY_FILE_LOCKING y -> n
 MFD_SEC_CORE y -> n
 PRISM54 n -> m
 RAW_DRIVER m -> n
 SND_SOC_INTEL_SKYLAKE_HDAUDIO_CODEC n -> y
 SND_SOC_ZX_AUD96P22 m -> n
 TEST_SORT n -> m
 UBSAN n -> y
 VFIO_MDEV_DEVICE m -> n
+CC_HAS_UBSAN_BOUNDS y
+MFD_TPS68470 n
+TEST_UBSAN n
+UBSAN_BOOL n
+UBSAN_BOUNDS y
+UBSAN_DIV_ZERO n
+UBSAN_ENUM n
+UBSAN_ONLY_BOUNDS y
+UBSAN_SANITIZE_ALL y
+UBSAN_SHIFT n
+UBSAN_TRAP y
```

</details>

**Comment by qzed** [comment]
> Unfortunately I think a binary search on the config switches is not possible

With that limited set of changes, we can do something fairly similar:  Use the working config, keep half of the options in the diff as they are and change the other half according to the diff. If the resulting kernel still works use its config for the next diff and so on.

**Comment by OliverUrbann** [comment]
Ok I got it. I did a binary search on the kernel flags using the git kernel repo mentioned above and version 5.13.13. It's:

```
CONFIG_ARCH_HAS_UBSAN_SANITIZE_ALL=y
CONFIG_UBSAN=y
CONFIG_UBSAN_TRAP=y
CONFIG_CC_HAS_UBSAN_BOUNDS=y
CONFIG_UBSAN_BOUNDS=y
CONFIG_UBSAN_ONLY_BOUNDS=y
# CONFIG_UBSAN_SHIFT is not set
# CONFIG_UBSAN_DIV_ZERO is not set
# CONFIG_UBSAN_BOOL is not set
# CONFIG_UBSAN_ENUM is not set
CONFIG_UBSAN_SANITIZE_ALL=y
# CONFIG_TEST_UBSAN is not set
```

I rechecked this by using the 5.14.9-surface config on 5.14.9 of the mainline kernel and commenting this block out makes this kernel working. 

So, what's next? Now it's time for a bug report? I'm wondering why this was switched on for 5.14 by Ubuntu. But on the other hand this should not lead to a frozen system on s2idle. So maybe a bug report where the developer for this switches are active?

Additionally, it would help me if you could patch them out before compiling the surface kernels till this is fixed some day, then I could update.

**Comment by qzed** [comment]
Nice work! In retrospect it kinda makes sense and I don't think it's an UBSAN bug: `CONFIG_UBSAN_TRAP=y` should be the cause for the crash. Help text for that is "On Sanitizer warnings, abort the running kernel code", i.e. it will crash if the kernel detects any undefined behavior.

I've disabled UBSAN in the config as I don't think it should normally be enabled (due to runtime costs etc.). A new kernel is building and should be available later today (as v5.14.10-2).

To track down the actual issue: Can you keep UBSAN enabled but set `CONFIG_UBSAN_TRAP=n`? This will hopefully give us a warning message (instead of the crash) indicating where the kernel detects undefined behavior (although you might need to set `no_console_suspend` or something like that). We'll then hopefully be able to fix that undefined behavior.

**Comment by k4ever** [comment]
I'm on a Core-i5 Surface Pro 4 running KDE Neon 5.22 (Ubuntu 20.04). I noticed that wake from suspend is broken and stopped working on ALL of the 5.14 kernels builds. If I put my tablet in suspend on 5.14.x, then press the power button to wake it, it will NEVER wake up! Works reliably every time with 5.13.13. 

I tested the new 5.14.10 build this morning and the problem is still there. Will there be a separate 5.14.10-2 build released today or is the 5.14.10 build that I just installed less than 30 minutes ago it for today?

Sorry for coming to the party late. I've been extremely busy in school (I still am), so I just set my default kernel to 5.13.13 and left it at that.

**Comment by qzed** [comment]
> 

> ```
> 
> ```
> 
> 
>     
>   
> 
> [There is](https://lore.kernel.org/lkml/20210918150130.GA30982@xsang-OptiPlex-9020/T/#m68500c72614511b43fed58f5e282d9c5ffafca85) already a bug report (at least it looks like this, right?)

That looks like it, yeah.
